common/models/user.js

```diff
@@ -301,6 +301,22 @@ module.exports = function(User) {
     return fn.promise;
   };
 
+  User.observe('before delete', function(ctx, next) {
+    var AccessToken = ctx.Model.relations.accessTokens.modelTo;
+    ctx.Model.find({ where: ctx.where }, function(err, list) {
+      if (err) return next(err);
+
+      var pkName = ctx.Model.definition.idName() || 'id';
+      var ids = list.map(function(u) { return u[pkName]; });
+      ctx.where = {};
+      ctx.where[pkName] = { inq: ids };
+
+      AccessToken.destroyAll({ userId: { inq: ids }}, next);
+      console.log('Deleted token for users ', ids);
```

Comment:
Please use `debug` instead of `console.log`, and also either 1) log "deleting tokens..." before calling destroyAll 2) or log from inside of the callback passed to destroyAll.